pkg/controller/build/schedule.go

```diff
@@ -63,7 +65,7 @@ func (action *scheduleAction) Handle(ctx context.Context, build *v1.Build) (*v1.
 		return nil, err
 	} else if !allowed {
 		// Build not allowed at this state (probably max running builds limit exceeded) - let's requeue the build
-		return nil, nil
+		return build, nil
```

Comment:
Reading the code comment below, it seems we must store the status of the build into this critical section, instead of letting it to be done by the controller. We may extend the `toPendingPhase` to store any generic condition and state before returning to the controller in general.